sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyUtils.java

```diff
@@ -97,10 +95,12 @@ public static void changeHeaders(HttpRequest request, String xRecordingId, Strin
             URL originalUrl = originalUrlBuilder.toUrl();
 
             HttpHeaders headers = request.getHeaders();
+            if (upstreamUri == null) {
+                headers.add("x-recording-upstream-base-uri", originalUrl.toString());
+                headers.add("x-recording-mode", mode);
+                headers.add("x-recording-id", xRecordingId);
```

Comment:
I believe these should use `headers.set` as add will create a list of values if they are added multiple times.